src/coreclr/src/jit/morph.cpp

```diff
@@ -6111,7 +6111,8 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
         bool addExplicitNullCheck = false;
 
         // Implicit byref locals are never null.
-        if (!((objRef->gtOper == GT_LCL_VAR) && lvaIsImplicitByRefLocal(objRef->AsLclVarCommon()->GetLclNum())))
+        if (fgAddrCouldBeNull(objRef) &&
+            !((objRef->gtOper == GT_LCL_VAR) && lvaIsImplicitByRefLocal(objRef->AsLclVarCommon()->GetLclNum())))
```

Comment:
Doesn't `fgAddrCouldBeNull` handle the implicit byref case? So we could simplify this futher?